lib/geoengineer/resources/aws/api_gateway/aws_api_gateway_method.rb

```diff
@@ -33,12 +47,23 @@ class GeoEngineer::Resources::AwsApiGatewayMethod < GeoEngineer::Resource
 
   def to_terraform_state
     tfstate = super
-    tfstate[:primary][:attributes] = {
-      "rest_api_id" => _rest_api._terraform_id,
-      "resource_id" => _resource._terraform_id,
-      "http_method" => http_method,
-      "authorization" => authorization
-    }
+    tfstate[:primary][:attributes] =
+      if self.authorizer_id && ['CUSTOM', 'COGNITO_USER_POOL'].include?(authorization)
```

Comment:
I also think it would be cleaner here if you added all the common resources initially (rest_api_id, resource_id, http_method, authorization), and then added authorizer_id if CUSTOM or COGNITO_USER_POOL authorizer types were included.